blog/2019-05-21-vue-remark/index.md

```diff
@@ -0,0 +1,137 @@
+        baseDir: './content/docs', // Where files are located (required)
+        pathPrefix: '/docs', // Route prefix
+        layout: './src/layouts/Doc.vue', // What layout to use
+        remark: {
+          // optional remark options and plugins
+        }
+      }
+    }
+  ]
+}
+```
+
+💥 **@gridsome/vue-remark** is designed to work as a drop-in replacement for **@gridsome/source-filesystem**. If you're already using **filesystem-source** you can change out the **use** and add **layout** and **pathPrefix** options.
+
+In this example any `.md` file in `./content/docs` will be a page.
+
+- `/content/docs/index.md` will be  **/docs**
+- `/content/docs/how-to-use.md` will be **/docs/how-to-use**
+- `/content/docs/folder/index.md` will be **/docs/folder**
+
+## Example Markdown file
+
+```jsx
+---
+title: A cool title
+excerpt: Lorem Ipsum is simply dummy text.
+---
+// Import Vue components
+import Youtube from '~/components/Youtube.vue'
+
+# {{ $frontmatter.title }}
+> {{ $frontmatter.excerpt }}
+
+Add **Markdown content** here...
+
+<Youtube id="xyxyxy" />
+
+Some more Markdown content here.
+```
+
+> All **front-matter** fields are available in `$frontmatter` and in the **GraphQL data layer** if you want to query data from other pages.
+
+
+### Use Markdown files as components.
+
+Use other `.md` files as Vue Components!
+
+```jsx
+---
+title: A cool title
+excerpt: Lorem Ipsum is simply dummy text.
+---
+import IntroBox from '~/components/IntroBox.md'
+
+<IntroBox :text="$frontmatter.excerpt" />
+```
+
+
+### Use any components that are registered globally.
+
+Register components globally in `src/main.js` if you don't want to import them for every page.
+Learn how to [make a component global](/docs/components#make-a-component-global).
+
+
+```jsx
+---
+title: A cool title
+---
+<GlobalComponent>
+  Global component
+</GlobalComponent>
+```
+
+### Use all the power of Vue Components in Markdown
+
+You can add GraphQL queries, script and style tags.
+
+```jsx
+---
+title: A cool title
+---
+
+Some _markdown_ content...
+
+<style lang="scss">
+.some-style {
+  border: red;
+}
+</style>
+
+<page-query>
+  ...
+</page-query>
+
+<script>
+  ...
+</script>
+```
+
+## What about MDX for Vue?
+MDX will have Vue.js support (https://mdxjs.com/guides/vue), but the problem is that it's still JSX. Under the hood **MDX** converts `.mdx` files into `JSX`, but **vue-remark**  simply converts `.md` files into `.vue` files. That's better and more efficient for Vue projects. 
+
+The MDX project is still great, so Gridsome will probably have an MDX plugin at some point also.
+
+
+## Gotchas...
+**Vue-remark** uses runs all files trough Webpack to render Vue Components. This will slow the build process down compared to for example **filesystem-source**. It's not recommended to use **vue-remark** if you plan to have thousands of blog posts.  **vue-remark** is perfect for documentation, design systems, portfolios, and smaller blogs.
```

Comment:
```suggestion
**Vue-remark** uses runs all files through Webpack to render Vue Components. This will slow the build process down compared to for example **filesystem-source**. It's not recommended to use **vue-remark** if you plan to have thousands of blog posts.  **vue-remark** is perfect for documentation, design systems, portfolios, and smaller blogs.
```